File: Algorithms/Easy/majority_element.py
from typing import List


class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        i = 0
        count = 0
        result = nums[0]
        while i < len(nums):
            if nums[i] == result:
                count += 1
            else:
                count -= 1
                if count < 0:
                    result = nums[i]
                    count = 0
            i += 1
        
        return result

File: Algorithms/Easy/majority_element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int count = 0;
        int result = nums.front();

        for (size_t i = 0; i != nums.size(); ++i) {
            if (nums[i] == result) {
                ++count;
            } else {
                --count;
                if (count < 0) {
                    result = nums[i];
                    count = 0;
                }
            }
        }

        return result;
    }
};
